File: backend/app/schemas/hotspot.py
from typing import Literal

from pydantic import BaseModel, Field


class HotspotData(BaseModel):
    id: str
    zone: str
    rank: int = Field(ge=1)
    risk: int = Field(ge=0, le=100)
    cluster: int = Field(ge=0)
    trend: Literal["up", "down"]
    change: float

File: frontend/src/types/hotspot.ts
/** Hotspot cluster from GET /api/hotspots */
export interface Hotspot {
  id: string;
  zone: string;
  rank: number;
  risk: number;
  cluster: number;
  trend: "up" | "down";
  change: number;
}

/** @deprecated Use Hotspot */
export type HotspotData = Hotspot;
